src/qmapshack/gis/CGisListWks.cpp

```diff
@@ -112,6 +112,7 @@ CGisListWks::CGisListWks(QWidget *parent)
     actionGroupSort->setExclusive(true);
     actionSortByTime    = addSortAction(this, actionGroupSort, "://icons/32x32/Time.png", tr("Sort by Time"), IGisProject::eSortFolderTime);
     actionSortByName    = addSortAction(this, actionGroupSort, "://icons/32x32/SortName.png", tr("Sort by Name"), IGisProject::eSortFolderName);
+    actionFilterProject = addAction(QIcon("://icons/32x32/Filter.png"), tr("Filter Project"), this, SLOT(slotAddProjectFilter()));
```

Comment:
I had that in one of the previous versions, however that was irritating, because it didn't adjust to the corresponding project.
Is there a way to achieve that?

Reply:
Don't you have already a method to add a `CProjectFilterItem` and to remove it again? And if you do not use the 'X' button in the search line to  remove it but to clear the line edit's content things should be pretty straight forward.